src/Microsoft.Health.Fhir.Shared.Client/FhirClient.cs

```diff
@@ -242,14 +242,24 @@ public Task<FhirResponse<Bundle>> SearchAsync(ResourceType resourceType, string
                 sb.Append("_count=").Append(count.Value);
             }
 
-            return SearchAsync(sb.ToString(), cancellationToken);
+            return SearchAsync(sb.ToString(), null, cancellationToken);
         }
 
         public async Task<FhirResponse<Bundle>> SearchAsync(string url, CancellationToken cancellationToken = default)
+        {
+            return await SearchAsync(url, null, cancellationToken);
+        }
+
+        public async Task<FhirResponse<Bundle>> SearchAsync(string url, Tuple<string, string> customHeader, CancellationToken cancellationToken = default)
```

Comment:
>Tuple<string, string> customHeader [](start = 72, length = 34)

I'm not quite sure I understand how this is related to custom search parameter update.
And how it's used in general?

Reply:
To use a newly created, or recently updated search parameter in a query string, before reindexing is complete, you must add a custom header to GET request.  By adding this header, I can test the updated search parameter without running a full reindex job.

Reply:
Only reason why I ask is I tried to find something in this PR what is using it, and I couldn't find it. Is it for future tests?

---
In reply to: [568010813](https://github.com/microsoft/fhir-server/pull/1615#discussion_r568010813) [](ancestors = 568010813)

Reply:
It is called from SearchTestsBase, which has a ExecuteAndValidateBundle method.  That method is called from the CustomSearchParamTests.